client/components/overview/OverviewContainer.jsx

```diff
@@ -20,9 +20,25 @@ const OverviewContainer = () => {
   const [productInfo, setProductInfo] = useState({});
 
   useEffect(() => {
-    getProductInfo(currentProduct, setProductInfo);
     getProductStyles(currentProduct, setProductStyles);
-  }, []);
+    getProductInfo(currentProduct, setProductInfo);
+  }, [currentProduct]);
+
+  useEffect(() => {
+    if (productStyles.length > 1) {
```

Comment:
What happens if `productStyles` is length one? If I'm understanding correctly, it looks like no initial style would be set in that case. It's an edge case to consider I think but great job getting the functionality to work!